examples/chip-tool-darwin/templates/commands.zapt

```diff
@@ -60,22 +62,31 @@ public:
         params.{{asStructPropertyName label}} = [NSNumber numberWith{{asObjectiveCNumberType "" type false}}:m{{asUpperCamelCase label}}];
         {{/if_chip_complex}}
         {{/chip_cluster_command_arguments}}
-        [cluster {{asLowerCamelCase name}}With{{#if (hasArguments)}}Params:params completionHandler:{{else}}CompletionHandler:{{/if}}
-        {{#if hasSpecificResponse}}
-            ^(CHIP{{asUpperCamelCase clusterName}}Cluster{{asUpperCamelCase responseName}}Params * _Nullable values, NSError * _Nullable error) {
-                 NSLog(@"Values: %@", values);
-        {{else}}
-            ^(NSError * _Nullable error) {
-        {{/if}}
-            chipError = [CHIPError errorToCHIPErrorCode:error];
-            ChipLogProgress(chipTool, "Error: %s", chip::ErrorStr(chipError));
-            SetCommandExitStatus(chipError);
-            }];
+        uint16_t repeatCount = mRepeatCount.ValueOr(1);
+        uint16_t __block responsesNeeded = repeatCount;
+        while (repeatCount--)
+        {
+            [cluster {{asLowerCamelCase name}}With{{#if (hasArguments)}}Params:params completionHandler:{{else}}CompletionHandler:{{/if}}
+            {{#if hasSpecificResponse}}
+                ^(CHIP{{asUpperCamelCase clusterName}}Cluster{{asUpperCamelCase responseName}}Params * _Nullable values, NSError * _Nullable error) {
+                    NSLog(@"Values: %@", values);
+            {{else}}
+                ^(NSError * _Nullable error) {
+            {{/if}}
+                chipError = [CHIPError errorToCHIPErrorCode:error];
+                responsesNeeded--;
+                if (chipError != CHIP_NO_ERROR) {
+                    ChipLogProgress(chipTool, "Error: %s", chip::ErrorStr(chipError));
+                }
+                if (responsesNeeded == 0) {
+                    SetCommandExitStatus(chipError);
```

Comment:
Same issue as in the non-generated version: we need to keep track of the errors we saw (chip-tool tracks the last one, but we just need to make sure that if there was an error we don't exit successfully, really).

Reply:
Updated